test/dt_arith.c

```diff
@@ -511,20 +521,21 @@ test_hard_query(void)
         goto error;
     }
 
-    /* Unregister the hard conversion from int to float.  Verify the conversion
-     * is a soft conversion. */
-    H5Tunregister(H5T_PERS_HARD, NULL, H5T_NATIVE_INT, H5T_NATIVE_FLOAT,
-                  (H5T_conv_t)((void (*)(void))H5T__conv_int_float));
```

Comment:
This test was previously casting `H5T__conv_int_float` from a library conversion function into an application conversion function. Since the two have different prototypes, the changes in this PR starting tripping over this. This appears to have worked in the past since the different between application conversion functions and library conversion functions is that application functions just have a DXPL parameter at they very end of the list.